diagrams/sequence-diagrams/authentication.md

```diff
@@ -8,52 +8,100 @@ sequenceDiagram
+        user ->> browser: login
+        browser ->> gw: HTTP POST
+      end
+
+      gw -->> browser: redirect to Auth0 callback
+
+      browser ->> auth0: HTTP GET callback
+
+    else all other email addresses
+      auth0 ->> ses: use SES API
+      ses ->> user: send one time code by email
+      auth0 -->> browser: redirect to code entry screen
+      browser ->> auth0: HTTP GET
+      auth0 -->> browser: code entry screen
+      browser -->> user: display code entry screen
+
+      user ->> user: access email
 
-    user ->> browser: provide one time code
-    browser ->> auth0: HTTP POST one time code
-    auth0 -->> browser: redirect to auth0/authorize/resume
-    browser ->> auth0: GET /authorize/resume
-    auth0 -->> browser: redirect to GOV.UK Forms with authorization code
+
+      loop Until valid code provided
+        user ->> browser: provide one time code
+        browser ->> auth0: HTTP POST one time code
+
+        auth0 ->> auth0: check code
+
+        opt OTP incorrect
+          auth0 -->> browser: Incorrect code page
+          note over user,browser: The code is not correct
+        end
+
+        opt Brute force protection
+          auth0 -->> browser: Account blocked page
+          note over user,browser: Your account has been blocked<br />after multiple consecutive login attempts
+        end
+      end
+    end
+
+      auth0 -->> browser: redirect to Auth0 resume
+      browser ->> auth0: HTTP GET
+
+      auth0 -->> browser: redirect to GOV.UK Forms with authorization code
 
     browser ->> forms: HTTP GET with authorization code
     forms ->> auth0: HTTP POST token request with authorization code
-    auth0 -->> forms: ID token, Access token
-    forms ->> auth0: GET JSON Web Key Set (JWKS) to verify tokens
-    auth0 -->> forms: Respond with JWK file
-    forms ->> forms: Verify tokens
+    auth0 -->> forms: ID token
     forms ->> forms: extract email address from ID token
     alt First time user has accessed GOV.UK Forms
         forms ->> forms: create record for user
     else Returning user
-        forms ->> forms: GET record of user
+        forms ->> forms: get record of user
     end
-    forms -->> browser: redirect to original requested URL: '/', add user id and current time to _forms_admin_session cookie
-    browser ->> forms: GET '/'
-    forms -->> browser: Forms homepage HTML
-    browser -->> user: display GOV.UK Forms
 
+    forms ->> forms: create user session
+    forms -->> browser: set session cookie, redirect
+    browser ->> browser: store session cookie
     note right of user: user is now authenticated
+    browser ->> forms: HTTP GET
+    forms ->> forms: get list of forms
+
+    alt User has Super Admin role and user has not authenticated via Google Workspace
+      forms -->> browser: forbidden
+      browser -->> user: access is denied
+      note right of user: user is not authorised to use the platform
+    else all other email addresses
```

Comment:
Change to `else Normal User or User has Super Admin role and has authenticated via Google Workspace`